### DISCUSSION TITLE
PythonTransform - Reading values form a list to activate scoring

### DISCUSSION THREAD

**Discussion by MilanNedoma** [bonsai-rx/bonsai#1716] [Q&A]
Good morning,

I would like to ask whether it is possible for a Python Transform node to read real-time values from a list that is directly connected to it.
Please see the basic logic below. I can´t really share the whole layout and bonsai file, however I am just curious whether something like this is even possible.

The idea behind this is that I have a list of three boolean values, being updated in real-time. When the first position in the list is True, I want to display 0, when the second is True, display 1 and display 2 when the third position is True.

I have tried multiple ways of making this work, including a way to only use standard nodes included in the basic packages of Bonsai, however, with no luck so far.

Thank you very much in advance and have a great rest of the day.

![1](https://github.com/bonsai-rx/bonsai/assets/140824061/b84748df-d434-49ea-a9be-908456182d31)
![2](https://github.com/bonsai-rx/bonsai/assets/140824061/9d6f52d6-194d-41e1-970f-8032336a84aa)

**Comment by bruno-f-cruz** [comment]
Please attach your workflow! Thanks

**Reply by MilanNedoma** [reply]
Please see the workflow attached below and thank you in advance!

[PythonTransform.zip](https://github.com/bonsai-rx/bonsai/files/14663819/PythonTransform.zip)

**Reply by bruno-f-cruz** [reply]
A few points.
1. The input to your python transform is a Tuple not a List. 
2. I think the following python script will do what you want:
```python
from System import Tuple


@returns(Tuple[int, int, int])
def process(value):
  _l = [1,2,3]
  for i, b in enumerate(value):
    if b == True:
      _l[i] = 1
    else:
      _l[i] = 0
  return Tuple.Create(_l[0], _l[1], _l[2])
```

3. Dont use python transform for these transformations. The logic in Bonsai is much simpler. You can unpack the tuple and convert each branch individually:
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/9eceff7c-9dad-45a4-a732-146044713239)


Where ExpressionTransform is: `it ? 1.0 : 0.0`. 

4. Or, you can create your own dynamic class on-the-fly:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/696a9ced-4c51-4018-9c6c-f2a0ffca9bc3)

```csharp
new(
Item1 ? 1.0 : 0.0 as Foo,
Item2 ? 1.0 : 0.0 as Bar,
Item3 ? 1.0 : 0.0 as Baz)

```
